core/services/relay/evm/config_poller.go

```diff
@@ -117,11 +162,14 @@ func (cp *configPoller) Replay(ctx context.Context, fromBlock int64) error {
 
 // LatestConfigDetails returns the latest config details from the logs
 func (cp *configPoller) LatestConfigDetails(ctx context.Context) (changedInBlock uint64, configDigest ocrtypes.ConfigDigest, err error) {
-	latest, err := cp.destChainLogPoller.LatestLogByEventSigWithConfs(ConfigSet, cp.addr, 1, pg.WithParentCtx(ctx))
+	latest, err := cp.destChainLogPoller.LatestLogByEventSigWithConfs(ConfigSet, cp.aggregatorContractAddr, 1, pg.WithParentCtx(ctx))
 	if err != nil {
-		// If contract is not configured, we will not have the log.
 		if errors.Is(err, sql.ErrNoRows) {
-			return 0, ocrtypes.ConfigDigest{}, nil
+			if cp.isConfigStoreAvailable() {
+				// Fallback to RPC call in case logs have been pruned and configStoreContract is available
+				return cp.callLatestConfigDetails(ctx)
+			}
+			err = nil // log not found means return zero config digest
```

Comment:
Falling back in to the other error logic to achieve this make me nervous about future changes. Would it be more clear and less brittle to return directly from here:
```suggestion
			// log not found means return zero config digest
			return 0, ocrtypes.ConfigDigest{}, nil
```